From 4e75647ffbdb8fcad0e70a47c82104bcc0602109 Mon Sep 17 00:00:00 2001
From: Vignesh Sridharan <vignsrid@qti.qualcomm.com>
Date: Thu, 21 May 2026 12:37:25 +0530
Subject: [PATCH] bwl/bpl: Add smd_ptk_mode and smd_default_dl_drain_time

Extend SMD policy parameter persistence to include two previously
missing fields:
- smd_ptk_mode: written as 'smd_ptk_mode' to both hostapd config
  and UCI wireless (smd_ptk_mode)
- smd_default_dl_drain_time: written as 'uhr_dl_drain_duration_tu'
  to hostapd config and as 'smd_dl_drain_time' to UCI wireless

Signed-off-by: Vignesh Sridharan <vignsrid@qti.qualcomm.com>
---
 .../beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp   | 14 ++++++++++++++
 framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp    |  2 ++
 2 files changed, 16 insertions(+)

diff --git a/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp b/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
index 221bff772..26b7b03f9 100644
--- a/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
+++ b/common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp
@@ -2089,11 +2089,17 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
                     conf.set_create_vap_value(vap_id, "smd_max_peer_apmlds",
                                               std::to_string(bss_it.smd_max_prepared_ap_mlds));
                     conf.set_create_vap_value(vap_id, "smd_type", std::to_string(bss_it.smd_type));
+                    conf.set_create_vap_value(vap_id, "smd_ptk_mode",
+                                              std::to_string(bss_it.smd_ptk_mode));
+                    conf.set_create_vap_value(vap_id, "uhr_dl_drain_duration_tu",
+                                              std::to_string(bss_it.smd_default_dl_drain_time));
                     LOG(DEBUG) << "Written SMD config to hostapd for BSS " << bss_it.bssid
                                << ": smd_identifier=" << bss_it.smd_identifier
                                << ", timeout=" << bss_it.smd_timeout
                                << ", dl_data_fwd=" << int(bss_it.smd_dl_forwarding)
                                << ", max_peer_apmlds=" << int(bss_it.smd_max_prepared_ap_mlds)
+                               << ", ptk_mode=" << int(bss_it.smd_ptk_mode)
+                               << ", dl_drain_time=" << int(bss_it.smd_default_dl_drain_time)
                                << ", type=" << int(bss_it.smd_type);
                 } else {
                     // Clear SMD parameters when not applicable
@@ -2103,6 +2109,8 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
                     conf.set_create_vap_value(vap_id, "smd_dl_data_fwd", "");
                     conf.set_create_vap_value(vap_id, "smd_max_peer_apmlds", "");
                     conf.set_create_vap_value(vap_id, "smd_type", "");
+                    conf.set_create_vap_value(vap_id, "smd_ptk_mode", "");
+                    conf.set_create_vap_value(vap_id, "uhr_dl_drain_duration_tu", "");
                 }
 
                 //required to update uci wireless
@@ -3884,11 +3892,17 @@ bool ap_wlan_hal_nl80211::create_new_hostapd_conf_file(std::string &ifname, son:
         conf.set_create_vap_value(vap_id, "smd_max_peer_apmlds",
                                   std::to_string(bss_conf.smd_max_prepared_ap_mlds));
         conf.set_create_vap_value(vap_id, "smd_type", std::to_string(bss_conf.smd_type));
+        conf.set_create_vap_value(vap_id, "smd_ptk_mode",
+                                  std::to_string(bss_conf.smd_ptk_mode));
+        conf.set_create_vap_value(vap_id, "uhr_dl_drain_duration_tu",
+                                  std::to_string(bss_conf.smd_default_dl_drain_time));
         LOG(DEBUG) << "Written SMD config to hostapd for new BSS " << bss_conf.bssid
                    << ": smd_identifier=" << bss_conf.smd_identifier
                    << ", timeout=" << bss_conf.smd_timeout
                    << ", dl_data_fwd=" << int(bss_conf.smd_dl_forwarding)
                    << ", max_peer_apmlds=" << int(bss_conf.smd_max_prepared_ap_mlds)
+                   << ", ptk_mode=" << int(bss_conf.smd_ptk_mode)
+                   << ", dl_drain_time=" << int(bss_conf.smd_default_dl_drain_time)
                    << ", type=" << int(bss_conf.smd_type);
     }
 
diff --git a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
index eae29a967..587f625ce 100644
--- a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
+++ b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
@@ -722,6 +722,8 @@ bool bpl_cfg_set_wifi_credentials(const std::string &iface,
                 options["smd_dl_data_fwd"]     = std::to_string(config.smd_dl_forwarding);
                 options["smd_max_peer_apmlds"] = std::to_string(config.smd_max_prepared_ap_mlds);
                 options["smd_type"]            = std::to_string(config.smd_type);
+                options["smd_ptk_mode"]        = std::to_string(config.smd_ptk_mode);
+                options["smd_dl_drain_time"]   = std::to_string(config.smd_default_dl_drain_time);
             }
 
             return options;
-- 
2.34.1

